src/NewBookingForm.js

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useEffect, useState } from "react";
+
+const NewBookingForm = ({ bookingsList, addNewBooking }) => {
+  const [firstName, setFirstName] = useState("");
+  const [surname, setSurname] = useState("");
+  const [email, setEmail] = useState("");
+  const [title, setTitle] = useState("");
+  const [checkInDate, setCheckInDate] = useState("");
+  const [checkOutDate, setCheckOutDate] = useState("");
+  const [minDate, setMinDate] = useState("");
+
+  const updateFirstName = (event) => {
+    setFirstName(event.target.value);
+  };
+  const updateSurname = (event) => {
+    setSurname(event.target.value);
+  };
+  const updateEmail = (event) => {
+    setEmail(event.target.value);
+  };
+  const updateTitle = (event) => {
+    setTitle(event.target.value);
+  };
+  const updateCheckInDate = (event) => {
+    setCheckInDate(event.target.value);
+  };
+  const updateCheckOutDate = (event) => {
```

Comment:
We have a lot of very similar looking functions here, could you think of anyway to generalise them and make it reusable for all of these fields?

Reply:
Thank you, Sam! I've already created one function that saves all input values inside the object and then pass the object to handleSubmit function.